subxt/examples/setup_reconnecting_rpc_client.rs

```diff
@@ -0,0 +1,117 @@
+
+#![allow(missing_docs)]
+
+use std::{sync::Arc, time::Duration};
+
+use futures::StreamExt;
+use subxt::backend::legacy::LegacyBackend;
+use subxt::backend::rpc::reconnecting_rpc_client::{Client, ExponentialBackoff};
+use subxt::backend::rpc::RpcClient;
+use subxt::{OnlineClient, PolkadotConfig};
+
+// Generate an interface that we can use from the node's metadata.
+#[subxt::subxt(runtime_metadata_path = "../artifacts/polkadot_metadata_small.scale")]
+pub mod polkadot {}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt::init();
+
+    // Create a new client with with a reconnecting RPC client.
+    let rpc = Client::builder()
+        // Reconnect with exponential backoff
+        //
+        // This API is "iterator-like" and we use `take` to limit the number of retries.
+        .retry_policy(
+            ExponentialBackoff::from_millis(100)
+                .max_delay(Duration::from_secs(10))
+                .take(3),
+        )
+        // There are other configurations as well that can be found at [`reconnecting_rpc_client::ClientBuilder`].
+        .build("ws://localhost:9944".to_string())
+        .await?;
+
+    // If you want to use the unstable backend with the reconnecting RPC client, you can do so like this:
+    //
+    // ```
+    // use subxt::backend::unstable::UnstableBackend;
+    // use subxt::OnlineClient;
+    //
+    // let (backend, mut driver) = UnstableBackend::builder().build(RpcClient::new(rpc.clone()));
+    // tokio::spawn(async move {
+    //     while let Some(val) = driver.next().await {
+    //         if let Err(e) = val {
+    //             eprintln!("Error driving unstable backend: {e}; terminating client");
+    //        }
+    //    }
+    // });
+    // let api: OnlineClient<PolkadotConfig> = OnlineClient::from_backend(Arc::new(backend)).await?;
+    // ```
+
+    let backend: LegacyBackend<PolkadotConfig> =
+        LegacyBackend::builder().build(RpcClient::new(rpc.clone()));
+
+    let api: OnlineClient<PolkadotConfig> = OnlineClient::from_backend(Arc::new(backend)).await?;
+
+    // Optionally print if the RPC client reconnects.
+    let rpc2 = rpc.clone();
+    tokio::spawn(async move {
+        loop {
+            rpc2.reconnect_started().await;
+            let now = std::time::Instant::now();
+            rpc2.reconnected().await;
+            println!(
+                "RPC client reconnection took `{}s`",
+                now.elapsed().as_secs()
+            );
+        }
+    });
+
+    // Run for at most 100 blocks and print a bunch of information about it.
+    //
+    // The subscription is automatically re-started when the RPC client has reconnected.
+    // You can test that by stopping the polkadot node and restarting it.
+    let mut blocks_sub = api.blocks().subscribe_finalized().await?.take(100);
+    // Track the last block to determine how many blocks that were missed when reconnecting.
+    let mut last_seen_block = None;
```

Comment:
Being just an example, I wonder if it's worth this extra complexity vs just printing block details and continuing if disconnected_will_reconnect error :)